src/main/scala/org/globalforestwatch/summarystats/afi/AFiDF.scala

```diff
@@ -28,17 +28,75 @@ object AFiDF extends SummaryDF {
         throw new IllegalArgumentException(s"Can't produce DataFrame for $id")
     }
 
-    summaryRDD
+    // Transform the input fields as described below in the comments, and also change
+    // rdd into (key, value) form to allow reducing by (list_id, location_id,
+    // gadm_id).
+    val keyValueRDD = summaryRDD
       .flatMap {
         case Valid(Location(fid, data)) =>
           data.stats.map {
-            case (dataGroup, data) =>
-              (rowId(fid), RowError.empty, dataGroup, data)
+            case (dataGroup, data) => {
+              val r = rowId(fid)
```

Comment:
I think just using the full word for the val name rather the first letter will make this easier to read